### ISSUE TITLE
The angle icon inside clr-accordion is decorative and needs to be ignored by screen readers.

### ISSUE THREAD

**Issue by lf-collins** [vmware/clarity#5319]
## Describe the accessibility issue

We got a report from AccessLevel where they note that the angle icon inside clr-accordion is decorative and needs to be ignored by screen readers.

## WCAG Guideline
[Issue - WCAG20]
The right and down arrow icon elements for the accordion headers are decorative and needs to be hidden. The textual equivalent is competing with the state (aria-expanded) of the buttons.
[Impact]
Decorative icons with textual equivalents that don't add value or meaning can only add to the verbosity when using screen reader.

## How to reproduce

https://clarity.design/documentation/accordion

<img width="1214" alt="Screen Shot 2020-11-19 at 1 21 27 PM" src="https://user-images.githubusercontent.com/26111301/99625547-75229500-2a6b-11eb-83f0-6695e199fa90.png">

## Expected behavior

When the icon is decorative and needs to be ignored by screen readers, add aria-hidden="true".

**Comment by gnomeontherun** [comment]
@chlane @scroniser Can you review when you have a chance?

**Comment by chlane** [comment]
aria-expanded should suffice to programmatically express the expanded/collapsed state of the control which makes the icon decorative in my opinion.

**Comment by lf-collins** [comment]
Another problem is very similar to this one, so I describe it here.

**How to reproduce:** https://clarity.design/documentation/input
The exclamation icon is marked as **aria-hidden=true**, but what it conveys is error.
**Expected behavior:** we should remove aria-hidden=true and add equivalent text, for example `<clr-icon aria-label="Error" title="Error" ...>`
<img width="1227" alt="Screen Shot 2020-11-20 at 4 18 41 PM" src="https://user-images.githubusercontent.com/26111301/99777107-cc4c6680-2b4c-11eb-9c9e-227d6416f84d.png">

AccessLevel report:
> [Issue - WCAG20]
The exclamation mark images convey meaning as errors but there are no textual equivalent to convey this using screen reader.
The meaning of the images is lost without textual equivalent when using screen reader and users will not be able to decipher that an error has occurred. 
[Recommendation]
Add textual equivalent to image especially if it conveys meaning and purpose. This can be done adding title element within svg elements. aria-label and/or title attribute is also acceptable although title attribute is not conveyed by screen reader when using virtual cursor and ARIA is dependent on AT and browser compatibility.
[Code Compliant Example]
<clr-icon aria-label="Error" title="Error" ...>

**Comment by chlane** [comment]
Users may not understand the exact text the icon is intended to stand for. In my opinion, icons should always be hidden from assistive technology and the text that they stand for should always be visible. I am aware that this makes the icons visually redundant and takes up more space. When icons are used for controls, speech recognition users need to verbally address these controls in a precise manner. Guessing the text that an icon represents can lead to inconsistent results. In the case of an error icon, one can argue that users understand that it stands for the text "error". But it could stand for the text "warning" or "alert". Since it is not an interactive control, this case is less severe. But it is simpler to never use icons to replace visual text.

**Comment by scroniser** [comment]
@chlane - I've flagged this as a Clarity Core issue. Yes, I concur, visible text is far favorable to labeling the icon. In instances where visible text is not used, labeling the icon will ensure all users have access to the icon. In instances where visible text isn't used, people who use screen readers are at an advantage.

**Comment by scroniser** [comment]
> aria-expanded should suffice to programmatically express the expanded/collapsed state of the control which makes the icon decorative in my opinion.

Thinking about this further, the button cannot be empty. The button must use a label so that it can be accessed with voice navigation.

**Comment by mathisscott** [comment]
> @chlane - I've flagged this as a Clarity Core issue. Yes, I concur, visible text is far favorable to labeling the icon. In instances where visible text is not used, labeling the icon will ensure all users have access to the icon. In instances where visible text isn't used, people who use screen readers are at an advantage.

Hi @scroniser 

The bug is being reported against Clarity Angular. This is not Clarity Core and the problem does not exist in Clarity Core.

**Comment by bdryanovski** [comment]
Can we summarise what is the best practice of having `aria-hidden` on icons. We had few issues adding/removing this attribute on places and I'm not sure when/where is the correct way of using it anymore. 

Even if we are talking only for Clarity Angular this could be easily fixable and consistent with Angular Core. 

@scroniser @chlane

**Comment by scroniser** [comment]
@bbogdanov it's best to leave the decision with the clarity consumer, allowing them to implement `aria-hidden` or good alt text. If that isn't possible, err on the side of caution -- use good alt text instead of `aria-hidden`
